eudemo/eudemo/maintenance/lower_port.py

```diff
@@ -0,0 +1,122 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+EU-DEMO Lower Port
+"""
+from dataclasses import dataclass
+from typing import Dict, Tuple, Type, Union
+
+import numpy as np
+
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.geometry.coordinates import Coordinates
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import make_polygon
+from bluemira.geometry.wire import BluemiraWire
+
+
+@dataclass
+class LowerPortDesignerParams(ParameterFrame):
+    """LowerPort ParameterFrame"""
+
+    lower_port_angle: Parameter[float]
+    divertor_padding: Parameter[float]
+
+
+class LowerPortDesigner(Designer):
+    """
+    Lower Port Designer
+
+    Notes
+    -----
+    Retrictions on angle is ±45° because there are no protections
+    keeping the port dimensions constant
+
+    """
+
+    param_cls: Type[ParameterFrame] = LowerPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame],
+        build_config: Dict,
+        divertor_xz: BluemiraFace,
+        x_wall_inner: float,
+        x_wall_outer: float,
+        x_extrema: float,
+    ):
+        super().__init__(params, build_config)
+        self.divertor_xz = divertor_xz
+        self.x_wall_inner = x_wall_inner
+        self.x_wall_outer = x_wall_outer
```

Comment:
I note that I can quite quickly make an invalid face by specifying silly values here.